packages/core/content-manager/server/services/uid.js

```diff
@@ -31,7 +31,7 @@ module.exports = ({ strapi }) => ({
 
     const possibleColisions = await query
       .findMany({
-        where: { [field]: { $contains: value } },
+        where: { [field]: { $eqi: value } },
```

Comment:
From what I understand about this function, it behaves as if you can have the same "uid" several times. So the use of `$contains` makes sense here.
For example, we may already have `my-wanted-uid` and `my-wanted-uid-1` in the database.
In this case, we want `query.findMany()` to find both occurrences of this "uid" and have this function return `my-wanted-uid-2`.

If we change the `$contains` with your `$eqi`, we will no longer be able to give an "incremental" uid.

Reply:
Maybe you can try using `$startsWith` instead ?